src/main/java/org/dependencytrack/policy/CwePolicyEvaluator.java

```diff
@@ -0,0 +1,94 @@
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright (c) Steve Springett. All Rights Reserved.
+ */
+package org.dependencytrack.policy;
+
+ import alpine.common.logging.Logger;
+ import org.apache.commons.collections4.CollectionUtils;
+ import org.dependencytrack.model.Component;
+ import org.dependencytrack.model.Policy;
+ import org.dependencytrack.model.PolicyCondition;
+ import org.dependencytrack.model.Vulnerability;
+ import org.dependencytrack.parser.common.resolver.CweResolver;
+
+ import java.util.ArrayList;
+ import java.util.Arrays;
+ import java.util.List;
+
+/**
+ * Evaluates the Common Weakness Enumeration of component vulnerabilities against a policy.
+ */
+public class CwePolicyEvaluator extends AbstractPolicyEvaluator {
+
+    private static final Logger LOGGER = Logger.getLogger(CwePolicyEvaluator.class);
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public PolicyCondition.Subject supportedSubject() {
+        return PolicyCondition.Subject.CWE;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public List<PolicyConditionViolation> evaluate(final Policy policy, final Component component) {
+        final List<PolicyConditionViolation> violations = new ArrayList<>();
+        final List<PolicyCondition> policyConditions = super.extractSupportedConditions(policy);
+        for (final Vulnerability vulnerability : qm.getAllVulnerabilities(component, false)) {
+            for (final PolicyCondition condition: policyConditions) {
+                LOGGER.debug("Evaluating component (" + component.getUuid() + ") against policy condition (" + condition.getUuid() + ")");
+                if (matches(condition.getOperator(), vulnerability.getCwes(), condition.getValue())) {
+                    violations.add(new PolicyConditionViolation(condition, component));
+                }
+            }
+        }
+        return violations;
+    }
+
+    public boolean matches(final PolicyCondition.Operator operator, final List<Integer> vulnerabilityCwes, final String conditionValue) {
+
+        if (conditionValue == null || vulnerabilityCwes == null || "*".equals(conditionValue.trim())) {
+            return true;
+        }
+        if (PolicyCondition.Operator.IS == operator) {
+            final Integer cweId = CweResolver.getInstance().parseCweString(conditionValue);
+            if (cweId != null) {
+                return vulnerabilityCwes.contains(cweId);
+            }
+        } else {
+            List<Integer> cweIdsToMatch = new ArrayList<>();
+            List<String> conditionCwes = Arrays.asList(conditionValue.split(","));
+            conditionCwes.replaceAll(String::trim);
+            conditionCwes.stream().forEach(cwe -> {
+                Integer id = CweResolver.getInstance().parseCweString(cwe);
+                if(id != null)
+                    cweIdsToMatch.add(id);
+            });
+            if (PolicyCondition.Operator.CONTAINS_ANY == operator) {
```

Comment:
Should there be a check here whether `conditionCwes` is empty? Perhaps we should log a warning in that case, because it indicates a misconfiguration?

Reply:
That would be case if user types in condition value as " , " or just space or empty. In that case, we return false. I've added a empty check on cweIdsToMatch.